Content.Server/Advertise/AdvertiseComponent.cs

```diff
@@ -1,144 +1,49 @@
-            {
-                throw new PrototypeLoadException($"{Owner} should have minWait greater than or equal to maxWait for {Name}Component.");
-            }
-
-            // Start timer at initialization, without MinWait bound
-            RefreshTimer(false);
-        }
-
-        protected override void OnRemove()
-        {
-            _cancellationSource.Cancel();
-            _cancellationSource.Dispose();
-
-            base.OnRemove();
-        }
-
-        /// <summary>
-        /// Say advertisement and restart timer.
-        /// </summary>
-        private void SayAndRefresh()
-        {
-            IRobustRandom random = IoCManager.Resolve<IRobustRandom>();
-            IChatManager chatManager = IoCManager.Resolve<IChatManager>();
-
-            // Say advertisement
-            chatManager.EntitySay(Owner, Loc.GetString(random.Pick(_advertisements)));
-
-            // Refresh timer to repeat cycle
-            RefreshTimer();
-        }
+        public int MaximumWait { get; } = 10 * 60;
 
         /// <summary>
-        /// Refresh cancellation token and spawn new timer with random wait between <see cref="MinWait"/>
-        /// and <see cref="MaxWait"/>.
-        /// <param name="minBound">
-        /// Whether <see cref="MinWait"/> should be used to have a minimum waiting time.
-        /// </param>
+        ///     The identifier for the advertisements pack prototype.
         /// </summary>
-        private void RefreshTimer(bool minBound = true)
-        {
-            // Generate new source
-            _cancellationSource.Cancel();
-            _cancellationSource.Dispose();
-            _cancellationSource = new CancellationTokenSource();
-
-            // Generate random wait time, then create timer
-            IRobustRandom random = IoCManager.Resolve<IRobustRandom>();
-            var wait = minBound ? random.Next(MinWait * 1000, MaxWait * 1000) : random.Next(MaxWait * 1000);
-            Owner.SpawnTimer(wait, SayAndRefresh, _cancellationSource.Token);
-        }
+        [DataField("pack", customTypeSerializer:typeof(PrototypeIdSerializer<AdvertisementsPackPrototype>))]
+        public string PackPrototypeId { get; } = string.Empty;
 
         /// <summary>
-        /// Pause the advertising until <see cref="Resume"/> is called.
+        ///     The next time an advertisement will be said.
         /// </summary>
-        public void Pause()
-        {
-            // Cancel current timer
-            _cancellationSource.Cancel();
-        }
+        [ViewVariables]
+        public TimeSpan NextAdvertisementTime { get; set; } = TimeSpan.Zero;
 
         /// <summary>
-        /// Resume the advertising after pausing.
+        ///     Whether the entity will say advertisements or not.
         /// </summary>
-        public void Resume()
-        {
-            // Restart timer, without minBound
-            RefreshTimer(false);
-        }
+        [ViewVariables(VVAccess.ReadWrite)]
```

Comment:
No reason for readwrite if it doesn't do anything IMO.

Reply:
But changing this actually disables the advertisements, it does *actually* do something...

Reply:
Ideally the system should track which ones are enabled so we don't have another disposals situation but I guess most of them would be anyway...